internal/blockchain/difficulty.go

```diff
@@ -212,20 +211,152 @@ func (b *BlockChain) calcNextRequiredDifficulty(prevNode *blockNode, newBlockTim
 	return nextDiffBits
 }
 
+// calcNextBlake3Diff calculates the required difficulty for the block
+// AFTER the passed previous block node based on the difficulty retarget rules
+// defined in DCP0011.
+func (b *BlockChain) calcNextBlake3Diff(prevNode *blockNode, blake3Anchor *blockNode) uint32 {
+	// Calculate the time and height deltas as the difference between the
+	// provided block and the blake3 anchor block.
+	//
+	// Notice that if the difficulty prior to the activation point were being
+	// maintained, this would need to be the timestamp and height of the parent
+	// of the blake3 anchor block (except when the anchor is the genesis block)
+	// in order for the absolute calculations to exactly match the behavior of
+	// relative calculations.
+	//
+	// However, since the initial difficulty is reset with the agenda, no
+	// additional offsets are needed.
+	timeDelta := prevNode.timestamp - blake3Anchor.timestamp
+	heightDelta := prevNode.height - blake3Anchor.height
```

Comment:
When `blake3Anchor == prevNode` (i.e. validating the first block where this is activated), `timeDelta == 0` and `heightDelta == 0`.

Reply:
Correct, and this ensures the required difficulty for the first block when the agenda activates is exactly the initial starting difficulty.